src/Razor/src/Microsoft.VisualStudio.RazorExtension/SyntaxVisualizer/SourceMappingAdornmentTaggerProvider.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.VisualStudio.Editor.Razor.Documents;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.RazorExtension.SyntaxVisualizer
+{
+    [Export(typeof(IViewTaggerProvider))]
+    [ContentType(RazorConstants.RazorLSPContentTypeName)]
+    [TagType(typeof(IntraTextAdornmentTag))]
+    internal sealed class SourceMappingAdornmentTaggerProvider : IViewTaggerProvider
+    {
+        private readonly IBufferTagAggregatorFactoryService _bufferTagAggregatorFactoryService;
+        private readonly Lazy<RazorCodeDocumentProvidingSnapshotChangeTrigger> _sourceMappingProjectChangeTrigger;
+
+        [ImportingConstructor]
+        public SourceMappingAdornmentTaggerProvider(IBufferTagAggregatorFactoryService bufferTagAggregatorFactoryService, Lazy<RazorCodeDocumentProvidingSnapshotChangeTrigger> sourceMappingProjectChangeTrigger)
+        {
+            _bufferTagAggregatorFactoryService = bufferTagAggregatorFactoryService;
+            _sourceMappingProjectChangeTrigger = sourceMappingProjectChangeTrigger;
+        }
+
+        public ITagger<T>? CreateTagger<T>(ITextView textView, ITextBuffer buffer) where T : ITag
+        {
+            if (textView == null)
```

Comment:
Nit 1: `ITextView textView!!`?
Nit 2: I personally really dislike brace-less if-statements.

Reply:
Nit 1: Good point. I started this branch before the `!!` change was available.
Nit 2: Sorry, Roslyn is rubbing off on me. I never used to either 😞